src/core_write.cpp

```diff
@@ -141,40 +142,41 @@ void ScriptToUniv(const CScript& script, UniValue& out, bool include_address)
     out.pushKV("asm", ScriptToAsmStr(script));
     out.pushKV("hex", HexStr(script));
 
-    std::vector<std::vector<unsigned char>> solns;
-    TxoutType type = Solver(script, solns);
-    out.pushKV("type", GetTxnOutputType(type));
-
+    TxoutType type;
     CTxDestination address;
-    if (include_address && ExtractDestination(script, address) && type != TxoutType::PUBKEY) {
+
+    if (ExtractDestination(script, type, address) && include_address && type != TxoutType::PUBKEY) {
         out.pushKV("address", EncodeDestination(address));
     }
+    out.pushKV("type", GetTxnOutputType(type));
 }
 
-void ScriptPubKeyToUniv(const CScript& scriptPubKey,
-                        UniValue& out, bool fIncludeHex)
+void ScriptPubKeyToUniv(const CScript& scriptPubKey, UniValue& out, bool fIncludeHex)
 {
     TxoutType type;
-    std::vector<CTxDestination> addresses;
-    int nRequired;
+    CTxDestination address;
 
     out.pushKV("asm", ScriptToAsmStr(scriptPubKey));
     if (fIncludeHex)
         out.pushKV("hex", HexStr(scriptPubKey));
 
-    if (!ExtractDestinations(scriptPubKey, type, addresses, nRequired) || type == TxoutType::PUBKEY) {
-        out.pushKV("type", GetTxnOutputType(type));
-        return;
+    // TODO: from v0.22 ("addresses" and "reqSigs" deprecated) this entire if statement should be removed
+    if (IsDeprecatedRPCEnabled("reqSigs")) { // cruft...
+        std::vector<CTxDestination> addresses;
+        int nRequired;
+        if (ExtractDestinations(scriptPubKey, type, addresses, nRequired) && type != TxoutType::PUBKEY) {
+            out.pushKV("reqSigs", nRequired);
+            UniValue a(UniValue::VARR);
+            for (const CTxDestination& addr : addresses) {
+                a.push_back(EncodeDestination(addr));
+            }
+            out.pushKV("addresses", a);
+        }
     }
-
-    out.pushKV("reqSigs", nRequired);
-    out.pushKV("type", GetTxnOutputType(type));
-
-    UniValue a(UniValue::VARR);
-    for (const CTxDestination& addr : addresses) {
-        a.push_back(EncodeDestination(addr));
+    if (ExtractDestination(scriptPubKey, type, address) && type != TxoutType::PUBKEY) {
+        out.pushKV("address", EncodeDestination(address));
     }
-    out.pushKV("addresses", a);
+    out.pushKV("type", GetTxnOutputType(type));
```

Comment:
Same

Reply:
I changed the order here so that when the deprecation period is over, we just simply delete the `if (fIncludeDeprecated) {` branch. It also cleaned things up a bit as we don't need to push `out.pushKV("type", GetTxnOutputType(type));` twice. The behavior is the same though

Reply:
Anyways, if you think this is a bad idea or doesn't belong in this PR, I have no problem undoing these changes and trying to stick to the original code as closely as possible - just lmk. Part of the reason I added some fuzz tests in this PR https://github.com/bitcoin/bitcoin/pull/20772 was to help build confidence since these two functions have to be touched in this PR either way